scripts/imgtool/image.py

```diff
@@ -267,13 +270,18 @@ def ecies_hkdf(self, enckey, plainkey):
             newpk = X25519PrivateKey.generate()
             shared = newpk.exchange(enckey._get_public())
         derived_key = HKDF(
-            algorithm=hashes.SHA256(), length=48, salt=None,
+            algorithm=hashes.SHA256(), length=self.hkdf_len, salt=self.hkdf_salt,
             info=b'MCUBoot_ECIES_v1', backend=default_backend()).derive(shared)
+        if self.hkdf_salt is not None:
+            key_nonce = derived_key[48:64]
+            self.enc_nonce = derived_key[64:76] + bytes([0] * 4)
```

Comment:
While you are deriving 48 + 32 bytes of key material, here you only use 12 of the 16 remaining bytes (plus 4 zero-pad bytes), so 4 bytes are never consumed from the hkdf, why is that?

Reply:
We use additional 32 bytes from key material as:
- 16 bytes for key encryption IV
- 16 bytes for CTR image encryption IV (where 4 bytes are initially zero)

Reply:
This IVs generation was made to be consistently with AES data block size (16 bytes)